Source/Fuse.Reactive.JavaScript/Tests/ReactiveModules.uno

```diff
@@ -0,0 +1,66 @@
+using Uno;
+using Uno.Collections;
+using Uno.UX;
+using Uno.Testing;
+
+using Fuse.Controls;
+using Fuse.Navigation;
+using FuseTest;
+
+namespace Fuse.Reactive.Test
+{
+	public class ReactiveModulesTest : TestBase
+	{
+		[Test]
+		public void Basics()
+		{
+			var e = new UX.ReactiveModules();
+			var root = TestRootPanel.CreateWithChild(e);
+
+			root.StepFrameJS();
+			AssertList(e, "orange", "lemon", "banana", "coconut");
+			Assert.AreEqual(e.s.Value, "lemon");
+
+			e.change1.Perform();
+			root.StepFrameJS();
+			AssertList(e, "orange", "lemon", "banana", "coconut", "grape");
+			Assert.AreEqual(e.s.Value, "lemon");
+
+			e.change2.Perform();
+			root.StepFrameJS();
+			AssertList(e, "orange", "lemon", "lemon", "banana", "coconut", "grape");
+			Assert.AreEqual(e.s.Value, "lemon");
+
+			e.change3.Perform();
+			root.StepFrameJS();
+			AssertList(e, "orange", "lemon", "coconut", "grape");
+			Assert.AreEqual(e.s.Value, "lemon");
+
+			e.change4.Perform();
+			root.StepFrameJS();
+			AssertList(e, "orange", "fish", "coconut", "grape");
+			Assert.AreEqual(e.s.Value, "fish");
+
+			Assert.AreEqual(e.bar.Value, "3");
+			e.change5.Perform();
+			root.StepFrameJS();
+			Assert.AreEqual(e.bar.Value, "1");
+
+			e.change6.Perform();
+			root.StepFrameJS();
+			Assert.AreEqual(e.bar.Value, "8");
+			AssertList(e, "batman", "batman", "batman");
+			Assert.AreEqual(e.s.Value, "batman");
+
+			// Verify that dependent module was re-evaluated
+			e.change7.Perform();
+			root.StepFrameJS();
+		}
+
+		void AssertList(UX.ReactiveModules e, params string[] p)
```

Comment:
The `TestBase` class has convenience functions to make such testing easier, like `GetText`. Or you can use the `ObservableCollector` instead of `Each` for a more direct test.